Solutions/SAP BTP/Analytic Rules/BTP - Failed access attempts across multiple BAS subaccounts.yaml

```diff
@@ -0,0 +1,44 @@
+id: 74b243a6-3046-48aa-8b03-e43b3c529cc1
+kind: Scheduled
+name: BTP - Failed access attempts across multiple BAS subaccounts
+description: Identifies failed Business Application Studio access attempts over a
+  predefined number of subaccounts.
+severity: Medium
+status: Available
+requiredDataConnectors:
+  - connectorId: SAPBTPAuditLog
+    dataTypes:
+      - SAPBTPAuditLog_CL
+queryFrequency: 15m
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Reconnaissance
+  - Discovery
+relevantTechniques:
+  - T1595
+  - T1526
+query: |
+  let subaccount_detection_threshold = 3;
+  SAPBTPAuditLog_CL
+  | where Category == "audit.security-events" and Message has "Unauthorized access attempt"
+  | summarize Start=min(UpdatedOn), End=max(UpdatedOn), Tenants = make_set(Tenant) by UserName
```

Comment:
Provide maximum limit of make_set(). This will help with the performance of the query.

Reply:
Added limit to make_set()